benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/scheduler/ExperimentDispatcher.java

```diff
@@ -0,0 +1,79 @@
+package cloud.benchflow.experimentmanager.scheduler;
+
+import cloud.benchflow.experimentmanager.BenchFlowExperimentManagerApplication;
+import cloud.benchflow.experimentmanager.models.BenchFlowExperimentModel;
+import cloud.benchflow.experimentmanager.services.external.BenchFlowTestManagerService;
+import cloud.benchflow.experimentmanager.services.internal.dao.BenchFlowExperimentModelDAO;
+import java.util.concurrent.BlockingQueue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Jesper Findahl (jesper.findahl@usi.ch) created on 2017-05-19
+ */
+public class ExperimentDispatcher implements Runnable {
+
+  private static Logger logger =
+      LoggerFactory.getLogger(ExperimentDispatcher.class.getSimpleName());
+
+  private final BlockingQueue<String> readyQueue;
+  private final BlockingQueue<String> runningQueue;
+
+  private BenchFlowExperimentModelDAO experimentModelDAO;
+  private BenchFlowTestManagerService testManagerService;
+  private ExperimentTaskScheduler taskScheduler;
+
+  public ExperimentDispatcher(BlockingQueue<String> readyQueue,
+      BlockingQueue<String> runningQueue) {
+    this.readyQueue = readyQueue;
+    this.runningQueue = runningQueue;
+
+    this.experimentModelDAO = BenchFlowExperimentManagerApplication.getExperimentModelDAO();
+    this.testManagerService = BenchFlowExperimentManagerApplication.getTestManagerService();
+  }
+
+  @Override
+  public void run() {
+
+    logger.info("running");
+
+    while (true) {
+
+      try {
+
+        // wait for next experiment to be ready
+        String experimentID = readyQueue.take();
```

Comment:
The experiment dispatcher runs in a separate thread so it should not block anything else than that thread (e.g. itself).